plugin.xml

```diff
@@ -16,18 +16,29 @@
         <clobbers target="window.bluetoothSerial" />
     </js-module>
 
-    <platform name="android">
+    <platform name="android"
+        kotlin="1.2.30">
+        <!-- Add Kotlin support -->
+        <hook type="after_plugin_add" src="hooks/support-kotlin.js" />
+        <hook type="after_platform_add" src="hooks/support-kotlin.js" />
+        <framework src="org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.2.31" />
+        <apply-plugin>kotlin-android-extensions</apply-plugin>
+        <apply-plugin>kotlin-kapt</apply-plugin>
+
+        <!-- Cleanup Kotlin source on uninstall -->
+        <hook type="before_plugin_rm" src="hooks/on-uninstall.js" />
+        <hook type="before_plugin_uninstall" src="hooks/on-uninstall.js" />
 
         <config-file target="res/xml/config.xml" parent="/widget">
             <feature name="BluetoothSerial">
                 <param name="android-package" value="com.megster.cordova.BluetoothSerial"/>
             </feature>
         </config-file>
 
-        <source-file src="src/android/com/megster/cordova/BluetoothSerial.java"
-            target-dir="src/com/megster/cordova"/>
-        <source-file src="src/android/com/megster/cordova/BluetoothSerialService.java"
-            target-dir="src/com/megster/cordova"/>
+        <source-file src="src/android/com/megster/cordova/BluetoothSerial.kt"
+            target-dir="app/src/main/java/com/megster/cordova"/>
```

Comment:
for some reason I couldn't get the previous relative paths to work - the kotlin files were being copied to a top level directory and then were not available in the classpath